x-pack/plugins/security_solution/server/endpoint/routes/policy/handlers.ts

```diff
@@ -64,3 +67,33 @@ export const getAgentPolicySummaryHandler = function (
     });
   };
 };
+
+export const getPolicyListHandler = function (
+  endpointAppContext: EndpointAppContext
+): RequestHandler<
+  undefined,
+  TypeOf<typeof GetEndpointPackagePolicyRequestSchema.query>,
+  undefined
+> {
+  return async (context, request, response) => {
+    const soClient = context.core.savedObjects.client;
+    const fleetServices = endpointAppContext.service.getScopedFleetServices(request);
+    const endpointFilteredKuery = `${
```

Comment:
I worked with @parkiino yesterday on this and we validated that what i thought was going to happen (retrieval of non-endpoint package policies) did not actually happen. that's because even if a user of the API ends their `kql` with ` 0R`, when we append the `AND package.name: endpoint` to it, you get a KQL parsing error (`...OR AND...`). So I think we're good, especially since parentheses were added around the request's `kuery` value (if any).